src/apps/filters/h3.c

```diff
@@ -140,35 +185,177 @@ H3Error cellToBoundaryCmd(int argc, char *argv[]) {
     return E_SUCCESS;
 }
 
-bool generalHelp(int argc, char *argv[]) {
-    Arg *args[] = {&helpArg, &cellToLatLngArg, &latLngToCellArg,
-                   &cellToBoundaryArg};
+/// Inspection subcommands
 
-    const char *helpText =
-        "Please use one of the subcommands listed to perform an H3 "
-        "calculation. Use h3 <SUBCOMMAND> --help for details on the usage of "
-        "any subcommand.";
-    return parseArgs(argc, argv, sizeof(args) / sizeof(Arg *), args, &helpArg,
-                     helpText);
+SUBCOMMAND(getResolution, "Extracts the resolution (0 - 15) from the H3 cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&getResolutionArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // TODO: Should there be a general `isValidIndex`?
+    H3Error cellErr = H3_EXPORT(isValidCell)(cell);
+    H3Error edgeErr = H3_EXPORT(isValidDirectedEdge)(cell);
+    H3Error vertErr = H3_EXPORT(isValidVertex)(cell);
+    if (cellErr && edgeErr && vertErr) {
+        // Not exactly sure how to handle this
+        return cellErr | edgeErr | vertErr;
+    }
```

Comment:
No, because if it's a valid edge but not a valid vertex, then we still want to process the data because the index is an edge index and not a vertex index. So only if it fails *all* index types are we sure it's not a valid index.

Reply:
I see, in that case I think the `return cellErr || edgeErr || vertErr;` is still right.

Reply:
Well, that would always short-circuit to just `cellErr` in that case, since they'd all be non-zero, so why not just `return cellErr` in that case?

Reply:
Ok, fair enough. As long as a single value is returned I think it's ok.